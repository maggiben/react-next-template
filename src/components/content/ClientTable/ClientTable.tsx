import { 
  CheckCircleIcon, 
  TableView, 
  Column, 
  Cell, 
  CloseCircleIcon, 
  Label,
  Button,
} from '@fravega-it/bumeran-ds-fvg'
import { useTranslation } from 'react-i18next';
// import * as utils from '@utils/string';
import * as utils from '@utils/index';
import { Person } from 'types/type';
import styled from "styled-components";

const Center = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Left = styled.div`
  display: flex;
  justify-content: end;
  align-items: center;
`;

const Right = styled.div`
  display: flex;
  justify-content: start;
  align-items: center;
`;

type ClientTableProps = {
  person: Person;
}

// const booleanToText = (variable: boolean) => variable ? 'confirmado' : 'no confirmado'
// const capitalized = (word: string) => word.charAt(0).toUpperCase() + word.slice(1);

export default function ClientTable(props: ClientTableProps) {
  const { person } = props;
  const { t } = useTranslation();
  return (
    <TableView
      items={[person]}
      renderColumns={() => (
        <>
          <Column minWidth={50} label={t('name')} />
          <Column minWidth={50} label={t('lastname')} />
          <Column minWidth={40} label={t('birth date')}/>
          <Column minWidth={50} label={t('facephi')} />
          <Column minWidth={220} label={t('email')} />
          <Column minWidth={70} label={t('confirmed email')} />
          <Column minWidth={40} label={t('resend email')} />
          <Column minWidth={40} label={t('renaper')} />
          <Column minWidth={40} label={t('id expiration date')} />
          <Column minWidth={100} label={t('address')} />
        </>
      )}
      renderCells={({ id, name, lastname, faceapi, renaper, address, email, age, identification }) => (
        <>
          <Cell>{name}</Cell>
          <Cell>{lastname}</Cell>
          <Cell>
            <Left>
              {new Date(age).toLocaleDateString()}
            </Left>
          </Cell>
          <Cell>
            <Label leftIcon={faceapi ? <CheckCircleIcon size="s" /> : <CloseCircleIcon size="s"/> } label={utils.string.booleanToText(faceapi)} color={faceapi ? "green" : "red"}/>                  
          </Cell>
          <Cell>
            {email.address}
          </Cell>
          <Cell>
            <Label leftIcon={email.confirmed ? <CheckCircleIcon size="s" /> : <CloseCircleIcon size="s"/> } label={utils.string.booleanToText(email.confirmed)} color={email.confirmed ? "green" : "red"}/>
          </Cell>
          <Cell>
            <Center>  
              <Button id={id} size="s" label={t('resend')} disabled={email.confirmed} />
            </Center>
          </Cell>
          <Cell>
            <Label leftIcon={renaper.confirmed ? <CheckCircleIcon size="s" /> : <CloseCircleIcon size="s"/> } label={utils.string.booleanToText(renaper.confirmed)} color={renaper.confirmed ? "green" : "red"}/>
          </Cell>
          <Cell>
            <Left>
              {new Date(identification.expiration).toLocaleDateString()}
            </Left>
          </Cell>
          <Cell>{address?.street}</Cell>
        </>
      )}
    />
  );
}