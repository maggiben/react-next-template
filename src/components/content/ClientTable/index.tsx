import { 
  CheckCircleIcon, 
  TableView, 
  Column, 
  Cell, 
  CloseCircleIcon, 
  Label,
  Button,
} from '@fravega-it/bumeran-ds-fvg'

import { Person } from '../index';

type ClientTableProps = {
  person: Person;
}
  
export default function ClientTable(props: ClientTableProps) {
  const { person } = props;
  return (
    <TableView
      items={[person]}
      renderColumns={() => (
        <>
          <Column minWidth={50} label="name" />
          <Column minWidth={100} label="faceapi" />
          <Column minWidth={70} label="email" />
          <Column minWidth={70} label="email confirmed" />
          <Column minWidth={40} label="resend email" />
          <Column minWidth={100} label="city" />
        </>
      )}
      renderCells={({ id, name, faceapi, email, city }) => (
        <>
          <Cell>{name}</Cell>
          <Cell>
            <Label leftIcon={faceapi ? <CheckCircleIcon size="s" /> : <CloseCircleIcon size="s"/> } label={faceapi.toString()} color={faceapi ? "green" : "red"}/>                  
            </Cell>
          <Cell>{email.address}</Cell>
          <Cell>
            <Label leftIcon={email.confirmed ? <CheckCircleIcon size="s" /> : <CloseCircleIcon size="s"/> } label={email.confirmed.toString()} color={email.confirmed ? "green" : "red"}/>
          </Cell>
          <Cell>
            {(function () {
                if(!email.confirmed) {
                  return <Button id={id} size="s" label="Resend" />
                }
              })()
            }
          </Cell>
          <Cell>{city}</Cell>
        </>
      )}
    />
  );
}