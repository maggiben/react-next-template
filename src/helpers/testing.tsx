import { default as baseRenderer } from 'react-test-renderer';
import { render, RenderResult } from '@testing-library/react';
import i18n from '@utils/i18n';
import { I18nextProvider } from 'react-i18next';
import { RecoilRoot } from 'recoil';
import { ThemeProvider } from 'styled-components';
import { theme, IDefaultTheme } from '@fravega-it/bumeran-ds-fvg';
import React, { ReactElement } from 'react';
import '@testing-library/jest-dom'

/**
 * Helper used to render components for testing.
 * The function appends the theme to the children.
 * */
interface IRenderer {
  create: (children: ReactElement) => RenderResult;
  baseRenderer: (children: ReactElement) => baseRenderer.ReactTestRenderer;
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme) => RenderResult;
  renderWithI18n: (children: ReactElement) => RenderResult;
}

export const renderer: IRenderer = {
  create: (children: ReactElement): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  baseRenderer: (children: ReactElement): baseRenderer.ReactTestRenderer =>
    baseRenderer.create(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithI18n: (children: ReactElement): RenderResult => 
    render(
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </ThemeProvider>
  ),
};