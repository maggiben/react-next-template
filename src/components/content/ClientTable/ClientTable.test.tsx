import React from 'react';
import { renderer } from '@helpers/testing';
import 'jest-styled-components';
import ClientTable from './ClientTable';

import { expect } from '@jest/globals';

test('Check ClientCard works', () => {
    const person = {
      id: '0001',
      name: 'Steve',
      age: 45,
      identification: {
        type: 'dni',
        number: 3000000
      },
      faceapi: true,
      email: {
        address: 'steve@fravega.com.ar',
        confirmed: true
      },
      profession: 'Doctor',
      city: 'Buenos Aires',
      cp: '1414',
      selected: false,
      status: {
        label: 'Ok',
        color: 'green',
      }
    }
    const tree = renderer.create(<ClientTable person={person} />).toJSON();
    expect(tree).toMatchSnapshot();
});
