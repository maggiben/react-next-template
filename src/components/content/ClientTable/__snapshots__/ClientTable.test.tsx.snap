// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check ClientCard works 1`] = `
.c14 {
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-family: Roboto;
  color: inherit;
}

.c11 {
  all: unset;
  display: block;
  font-size: 14px;
  line-height: 20px;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-family: Roboto;
  color: inherit;
}

.c4 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 600;
  font-family: Work Sans;
  color: inherit;
}

.c13 {
  fill: #105225;
}

.c12 {
  padding: 2px 12px 2px 8px;
  background-color: #E0FFEA;
  color: #105225;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 1000px;
}

.c10 {
  padding: 8px 12px;
}

.c2 {
  font-weight: unset;
  min-width: 50px;
  text-align: unset;
  background-color: #F0F0FA;
  height: 56px;
}

.c5 {
  font-weight: unset;
  min-width: 100px;
  text-align: unset;
  background-color: #F0F0FA;
  height: 56px;
}

.c6 {
  font-weight: unset;
  min-width: 70px;
  text-align: unset;
  background-color: #F0F0FA;
  height: 56px;
}

.c7 {
  font-weight: unset;
  min-width: 40px;
  text-align: unset;
  background-color: #F0F0FA;
  height: 56px;
}

.c3 {
  padding: 8px 12px;
}

.c0 {
  width: 100%;
  border-collapse: separate;
  border-spacing: 0;
}

.c0 tr th:last-child,
.c0 tr td:last-child {
  border-right: 0.5px solid #D5D5E7;
}

.c0 tr td,
.c0 tr th {
  border-bottom: 0.5px solid #D5D5E7;
}

.c0 tr th:first-child,
.c0 tr td:first-child {
  border-left: 0.5px solid #D5D5E7;
}

.c0 tr th {
  border-top: solid 0.5px #D5D5E7;
}

.c0 tr:first-child th:last-child {
  border-top-right-radius: 16px;
}

.c0 tr:first-child th:first-child {
  border-top-left-radius: 16px;
}

.c0 tr:last-child td:first-child {
  border-bottom-left-radius: 16px;
}

.c0 tr:last-child td:last-child {
  border-bottom-right-radius: 16px;
}

.c8 tr:hover {
  background: #FAFAFF;
}

.c9 {
  min-height: 56px;
  color: #242433;
}

.c1 {
  color: #242433;
}

<div
  className=""
>
  <table
    className="c0"
  >
    <thead
      className="c1"
    >
      <tr>
        <th
          className="c2"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="c3"
          >
            <div
              aria-level={6}
              className="c4"
              color="inherit"
              role="heading"
              size="xs"
            >
              name
            </div>
          </div>
        </th>
        <th
          className="c5"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="c3"
          >
            <div
              aria-level={6}
              className="c4"
              color="inherit"
              role="heading"
              size="xs"
            >
              faceapi
            </div>
          </div>
        </th>
        <th
          className="c6"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="c3"
          >
            <div
              aria-level={6}
              className="c4"
              color="inherit"
              role="heading"
              size="xs"
            >
              email
            </div>
          </div>
        </th>
        <th
          className="c6"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="c3"
          >
            <div
              aria-level={6}
              className="c4"
              color="inherit"
              role="heading"
              size="xs"
            >
              email confirmed
            </div>
          </div>
        </th>
        <th
          className="c7"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="c3"
          >
            <div
              aria-level={6}
              className="c4"
              color="inherit"
              role="heading"
              size="xs"
            >
              resend email
            </div>
          </div>
        </th>
        <th
          className="c5"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="c3"
          >
            <div
              aria-level={6}
              className="c4"
              color="inherit"
              role="heading"
              size="xs"
            >
              city
            </div>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      className="c8"
    >
      <tr
        className="c9"
      >
        <td
          className="c10"
        >
          <p
            className="c11"
            color="inherit"
            size="m"
          >
            Steve
          </p>
        </td>
        <td
          className="c10"
        >
          <p
            className="c11"
            color="inherit"
            size="m"
          >
            <div
              className="c12"
              color="green"
              size="m"
            >
              <svg
                className="c13"
                color="green"
                fill="none"
                viewBox="0 0 28 28"
                width="20px"
              >
                <path
                  clipRule="evenodd"
                  d="M14 7.776a6.224 6.224 0 1 0 0 12.448 6.224 6.224 0 0 0 0-12.448ZM6.22 14a7.78 7.78 0 1 1 15.56 0 7.78 7.78 0 0 1-15.56 0Zm10.664-2.106a.778.778 0 0 1 0 1.1l-3.112 3.112a.778.778 0 0 1-1.1 0l-1.556-1.556a.778.778 0 0 1 1.1-1.1l1.006 1.006 2.562-2.562a.778.778 0 0 1 1.1 0Z"
                  fillRule="evenodd"
                />
              </svg>
              <span
                className="c14"
                color="inherit"
                size="m"
              >
                true
              </span>
            </div>
          </p>
        </td>
        <td
          className="c10"
        >
          <p
            className="c11"
            color="inherit"
            size="m"
          >
            steve@fravega.com.ar
          </p>
        </td>
        <td
          className="c10"
        >
          <p
            className="c11"
            color="inherit"
            size="m"
          >
            <div
              className="c12"
              color="green"
              size="m"
            >
              <svg
                className="c13"
                color="green"
                fill="none"
                viewBox="0 0 28 28"
                width="20px"
              >
                <path
                  clipRule="evenodd"
                  d="M14 7.776a6.224 6.224 0 1 0 0 12.448 6.224 6.224 0 0 0 0-12.448ZM6.22 14a7.78 7.78 0 1 1 15.56 0 7.78 7.78 0 0 1-15.56 0Zm10.664-2.106a.778.778 0 0 1 0 1.1l-3.112 3.112a.778.778 0 0 1-1.1 0l-1.556-1.556a.778.778 0 0 1 1.1-1.1l1.006 1.006 2.562-2.562a.778.778 0 0 1 1.1 0Z"
                  fillRule="evenodd"
                />
              </svg>
              <span
                className="c14"
                color="inherit"
                size="m"
              >
                true
              </span>
            </div>
          </p>
        </td>
        <td
          className="c10"
        >
          <p
            className="c11"
            color="inherit"
            size="m"
          />
        </td>
        <td
          className="c10"
        >
          <p
            className="c11"
            color="inherit"
            size="m"
          >
            Buenos Aires
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
